### ISSUE TITLE
Install distributed locking to the startup templates with proper configurations

### ISSUE THREAD

**Issue by hikalkan** [abpframework/abp#13761]
Application startup template has different options. In some options (e.g. separate auth server, tiered, ...) we are installing Redis packages and configuring it by default.
I think we can configure distributed locking too, because some ABP components are using it and we should configure it for [clustered deployment](https://docs.abp.io/en/abp/latest/Deployment/Clustered-Environment#configuring-a-distributed-lock-provider).

So, basically, we should install and configure distributed locking with redis provider ([see documentation](https://docs.abp.io/en/abp/latest/Distributed-Locking#configuring-a-provider) to learn how to do) for each application that we configure Redis, in the startup templates.
